Model/Indexer/Data.php

```diff
@@ -61,37 +64,48 @@ class Data implements IndexerActionInterface, MviewActionInterface
     /** @var NostoHelperAccount */
     private $nostoHelperAccount;
 
+    /** @var NostoLogger */
+    private $nostoLogger;
+
     /**
      * @param NostoIndexService $nostoServiceIndex
      * @param IndexCollectionFactory $indexCollectionFactory
      * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoLogger $nostoLogger
      */
     public function __construct(
         NostoIndexService $nostoServiceIndex,
         IndexCollectionFactory $indexCollectionFactory,
-        NostoHelperAccount $nostoHelperAccount
+        NostoHelperAccount $nostoHelperAccount,
+        NostoLogger $nostoLogger
     ) {
         $this->nostoServiceIndex = $nostoServiceIndex;
         $this->indexCollectionFactory = $indexCollectionFactory;
         $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->nostoLogger = $nostoLogger;
     }
 
     /**
      * @inheritdoc
-     * @throws \Exception
+     * @throws Exception
      */
     public function executeFull()
     {
         $storesWithNosto = $this->nostoHelperAccount->getStoresWithNosto();
         foreach ($storesWithNosto as $store) {
             $indexCollection = $this->getCollection($store);
-            $this->nostoServiceIndex->handleDirtyProducts($indexCollection, $store);
+            try {
+                $this->nostoServiceIndex->handleDirtyProducts($indexCollection, $store);
+            } catch (MemoryOutOfBoundsException $e) {
+                $this->nostoLogger->error($e->getMessage());
+                throw $e;
```

Comment:
How does the indexer behave if we throw the exception here? We don't want to terminate the other index processing. Maybe just log the exception and not throw anything.